.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,32 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+Thank you for submiting a contribution to Apache NiFi.
+
+In order to streamline the review of the contribution we ask you
+to ensure the follow steps have been taken:
+
+### For all changes:
+- [] Is there a JIRA ticket associated with this PR? Is is referenced 
+     on the commit message?
+
+### For code changes:
+- [] Ensure that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder.
+- [] Please write or update unit tests to verify your changes.
+- [] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under ASF 2.0?  Did you update the LICENSE file?
```

Comment:
We should also request that the initial PR is a single, squashed commit unless there is a good reason for doing so otherwise.